crates/sui-sdk/src/crypto.rs

```diff
@@ -80,15 +77,14 @@ impl AccountKeystore for FileBasedKeystore {
             .try_sign(msg)
     }
 
-    fn add_key(&mut self, keypair: AccountKeyPair) -> Result<(), anyhow::Error> {
-        let address: SuiAddress = keypair.public().into();
-        self.keys.insert(address, keypair);
+    fn add_key(&mut self, keypair: KeyPair) -> Result<(), anyhow::Error> {
+        self.keys.insert(keypair.public().inner_into(), keypair);
         self.save()?;
         Ok(())
     }
 
-    fn keys(&self) -> Vec<AccountPublicKey> {
-        self.keys.values().map(|key| key.public().clone()).collect()
+    fn keys(&self) -> Vec<PublicKey> {
```

Comment:
To be followed by a PR that uses this enum change (which might also require some refactoring on what we should call `AccountPublicKey`